src/main/java/com/ldtteam/structurize/client/gui/WindowReplaceBlock.java

```diff
@@ -124,38 +140,31 @@ public void onOpened()
         findPaneOfTypeByID("resourceIconTo", ItemIcon.class).setItem(new ItemStack(Blocks.AIR));
         findPaneOfTypeByID("resourceNameTo", Label.class).setLabelText((IFormattableTextComponent) new ItemStack(Blocks.AIR).getDisplayName());
         updateResources();
+        updateResourceList();
     }
 
     private void updateResources()
     {
         allItems.clear();
         allItems.addAll(ImmutableList.copyOf(StreamSupport.stream(Spliterators.spliteratorUnknownSize(ForgeRegistries.ITEMS.iterator(), Spliterator.ORDERED), false)
+            .filter(item -> item instanceof AirItem || item instanceof BlockItem || (item instanceof BucketItem && ((BucketItem) item).getFluid() != Fluids.EMPTY))
```

Comment:
Again this feels really like a code smell with the Air check.....

Reply:
AirItem is BlockItem, ask mojang why..